src/remote.c

```diff
@@ -30,7 +30,7 @@
 
 static int dwim_refspecs(git_vector *out, git_vector *refspecs, git_vector *refs);
 static int lookup_remote_prune_config(git_remote *remote, git_config *config, const char *name);
-char *apply_insteadof(git_config *config, const char *url, int direction);
+char *apply_insteadof(git_config *config, const char *url, int direction, int *matched);
```

Comment:
Generally we have out params as the first parameters, not the last.  And I'd use a boolean type here:

```suggestion
char *apply_insteadof(bool *matched, git_config *config, const char *url, int direction);
```

Reply:
good point, thanks for the hint!
i've pushed an update and will look into the failing tests next.

Reply:
Thanks!  I'm really happy to see this functionality.  Thanks for tackling it.

Reply:
oh oops, i forgot to update the local variables. with that the tests pass on my end.

Reply:
> Thanks! I'm really happy to see this functionality. Thanks for tackling it.

that's very kind of you to say, thank you!
this project had been incredibly easy to get started working on :relaxed: